pandas/core/algorithms.py

```diff
@@ -630,6 +634,9 @@ def factorize(
         uniques, codes = safe_sort(
             uniques, codes, na_sentinel=na_sentinel, assume_unique=True, verify=False
         )
+    if not dropna and (codes == na_sentinel).any():
+        uniques = np.append(uniques, [None])
```

Comment:
These are typically `int64` right? If so just curious if coercing to dtype is going to cause performance regressions

Reply:
i think `uniques` should be typically `str`/`category`, `codes` are `int`. i will rethink of this issue. thanks!